enhancements/ingress/optional-ingress-hypershift.md

```diff
@@ -0,0 +1,320 @@
+
+### Non-Goals
+- Use a different way for enabling/disabling the ingress other than via [the cluster capability](/enhancements/installer/component-selection.md).
+- Fully implement the ingress capability for the standalone OpenShift:
+  - Make the OpenShift installer tolerate the absence of the default ingress controller.
+  - Make the cluster authentication operator tolerate the absence of the default ingress controller.
+  - Make the cluster monitoring operator tolerate the absence of the default ingress controller.
+- Disable the Route API or try to tolerate the absence of the default ingress controller.
+- Design the capability API for HyperShift.
+- Implement the ingress operator disabling in the HyperShift's control plane operator.
+
+## Proposal
+
+### User Stories
+
+- As a HyperShift engineer, I want to be able to disable the ingress operator on the clusters where the ingress is implemented differently so that the hosted control plane can become lighter.
+- As a HyperShift engineer, I want the cluster operators to tolerate the absence of the ingress operator to keep the clusters from being degraded.
+- As a HyperShift hosted cluster's user, I want to see the ingress capability as a known capability to bring the visibility to all the cluster actors (admins, users, other operators).
+- As a HyperShift hosted cluster's user, I want the default ingress controller to not occupy the cluster resources while the ingress traffic is routed by the cloud provider's load balancers.
+
+### Workflow Description
+
+The HyperShift hosted cluster's user wants to use an AWS ALB to route the traffic to its example application deployed on a HyperShift's hosted cluster.
+
+#### Terminology
+[HyperShift concepts and personas](https://hypershift-docs.netlify.app/reference/concepts-and-personas/).
+
+#### Disable the default ingress
+
+- Cluster service consumer requests a hosted cluster without the ingress capability using the HostedCluster API (**note**: the HostedCluster's capability API does not exist and it's out of scope of this enhancement).
+- HyperShift creates the hosted cluster.
+- HyperShift skips the deployment of the ingress operator on the hosted control plane.
+- HyperShift creates `ClusterVersion` without the ingress capability on the hosted cluster.
+- Hosted cluster's cluster version operator does not deploy the cluster ingress operator's payload (CRDs, RBACs, etc.).
+
+#### Setup AWS ALB as the new ingress
+
+- Cluster instance admin installs [AWS Load Balancer Operator](/enhancements/ingress/aws-load-balancer-operator.md).
+- Cluster instance admin sets up the default `AWSLoadBalancerController`.
+- Cluster instance user creates the `Ingress` resource for the default aws load balancer controller's ingress class.
+- AWS load balancer controller provisions an ALB on AWS.
+- Cluster instance admin can install [External DNS Operator](/enhancements/dns/externaldns-operator.md) to help cluster instance users set up convenient DNS names.
```

Comment:
The External DNS Operator does not support ingresses (yet): https://github.com/openshift/external-dns-operator/blob/0cd8bee31e64f3086f9a44e3f5e299a3a1111175/api/v1beta1/externaldns_types.go#L471-L478

Reply:
True, I mixed up things: at some point I wanted this example to use `Service` which would stick well with what ExternalDNS Operator supports now. Anyway that was just a minor hint which I added to tight all together. But that's actually the unnecessary info which slightly overloads the EP: removed this line.